File: backend_dotnet/fruit_api/fruit_api/Models/Supplier.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace fruit_api.Models;

[Table("Suppliers")]
public class Supplier
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    [Column("supplierId")]
    public string SupplierId { get; set; } = null!;

    [Required]
    [MaxLength(150)]
    [Column("supplierName")]
    public string SupplierName { get; set; } = string.Empty;

    [MaxLength(15)]
    [Column("phone")]
    public string? Phone { get; set; }

    [MaxLength(100)]  // THÊM DÒNG NÀY - Giới hạn 100 ký tự cho email
    [Column("email")]  // THÊM DÒNG NÀY - Tên cột trong database
    public string? Email { get; set; }  // THÊM DÒNG NÀY

    [MaxLength(255)]
    [Column("address")]
    public string? Address { get; set; }

    [MaxLength(20)]
    [Column("status")]
    public string Status { get; set; } = "active";

    [Column("createdAt")]
    public DateTime CreatedAt { get; set; } = DateTime.Now;

    // Navigation properties
    public ICollection<Product>? Products { get; set; }
}

File: frontend_flutter/fruit_market/lib/models/Supplier.dart
import 'product.dart';

class Supplier {
  final String supplierId;
  final String supplierName;
  final String? phone;
  final String? address;
  final String status;
  final DateTime createdAt;

  final List<Product>? products;

  Supplier({
    required this.supplierId,
    required this.supplierName,
    this.phone,
    this.address,
    required this.status,
    required this.createdAt,
    this.products,
  });

  /// JSON -> Object
  factory Supplier.fromJson(Map<String, dynamic> json) {
    return Supplier(
      supplierId: json['supplierId'] ?? '',
      supplierName: json['supplierName'] ?? '',
      phone: json['phone'],
      address: json['address'],
      status: json['status'] ?? 'active',
      createdAt: json['createdAt'] != null 
          ? DateTime.parse(json['createdAt']) 
          : DateTime.now(),
      products: json['products'] != null
          ? (json['products'] as List)
              .map((e) => Product.fromJson(e))
              .toList()
          : null,
    );
  }

  /// Object -> JSON
  Map<String, dynamic> toJson() {
    return {
      'supplierId': supplierId,
      'supplierName': supplierName,
      'phone': phone,
      'address': address,
      'status': status,
      'createdAt': createdAt.toIso8601String(),
      'products': products?.map((e) => e.toJson()).toList(),
    };
  }
}